### PR TITLE
Adds a new check for required field changes

### PR DESCRIPTION
**Pull request by varadarajan-tw**
As a followup to the recent Google Sheets SEV, this PR adds a new github actions check to flag when new required fields are added to a destination already on main branch.
This check just adds a warning. It is upto the reviewer and the author to decide to decide on what they want to do with this warning.

To achieve this 
- A new command has been added to `list-required-fields` has been added. It simply lists all required fields of a destination by action.
- `list-required-fields` is run on both main and feature branch.
- The diff in required fields between main and feature branch is computed and a comment is added on the PR.

image

image

## Testing

Testing completed successfully via CI.

### CODE REVIEW

File: .github/REQUIRED_FIELD_WARNING_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# New required fields detected
+
+> [!WARNING]
+> Your PR adds new required fields to existing destination. Adding new required settings/mappings for a destination already in production requires updating existing customer destination configuration. Ignore this warning if this PR is for a new destination with no active customers in production.
+
+The following required fields were added in this PR:
+
+{{FIELDS_ADDED}}
+
+Consider adding these new fields as optional instead and assume default values in `perform` or `performBatch` block.
```
Comment (brennan) [inline .github/REQUIRED_FIELD_WARNING_TEMPLATE.md:12]:
```suggestion
Add these new fields as optional instead and assume default values in `perform` or `performBatch` block.
```

Maybe rephrase as a command, since we don't support adding required fields now. The contributor will have to add the fields as optional.

File: .github/REQUIRED_FIELD_WARNING_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<!--REQUIRED_FIELD_DIFF-->
+
+# New required fields detected
+
+> [!WARNING]
+> Your PR adds new required fields to existing destination. Adding new required settings/mappings for a destination already in production requires updating existing customer destination configuration. Ignore this warning if this PR is for a new destination with no active customers in production.
```
Comment (brennan) [inline .github/REQUIRED_FIELD_WARNING_TEMPLATE.md:6]:
```suggestion
> Your PR adds new required fields to an existing destination. Adding new required settings/mappings for a destination already in production requires updating existing customer destination configuration. Ignore this warning if this PR is for a new destination with no active customers in production.
```

**Review by brennan — DISMISSED** [review]

**Review by itsarijitray — APPROVED** [review]
